src/nfv_test_api/v2/services/gnodeb.py

```diff
@@ -223,25 +223,32 @@ def node_status(self, identifier: str) -> Dict[str, Any]:
         # make sure the config exists
         gnb = self.get_one(identifier)
 
-        if identifier not in self.process_handler.processes:
-            raise NotFound(f"No gNodeB process found for nci {identifier}")
-
-        status: Dict[str, Any] = {}
+        status: Dict[str, Any] = {"pid": None, "status": {}}
         node_name = f"UERANSIM-gnb-{int(gnb.mcc)}-{int(gnb.mnc)}-{int(gnb.nci[:-1], base=0)}"  # type: ignore
         command = [
             "nr-cli",
             node_name,
             "--exec",
             "status",
         ]
-
-        status["pid"] = self.process_handler.processes[identifier].pid
-        stdout, stderr = self.host.exec(command)
-        if stderr:
-            raise NotFound(f"Failed to fetch gNodeB status: {stderr}")
-
-        # Parse the status response, it should be a yaml object
-        status["status"] = yaml.safe_load(stdout) or {}
+        if identifier in self.process_handler.processes:
+            return_code = self.process_handler.processes[identifier].poll()
+            if return_code is not None:
+                # if process is terminated, it means it has crashed
+                self.process_handler.kill(identifier)  # handle zombie process
```

Comment:
If it has terminated, why do we need to kill it?

Reply:
Actually the situation is a bit more complex, the process is terminated but still there as zombie process. So to summarize: If the process is still in `self.process_handler` but has terminated then it is a zombie process.

Reply:
I will add more comments to make it clear